ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Image/ImagePage.tsx

```diff
@@ -93,8 +96,9 @@ function ImagePage() {
 
     const pagination = useURLPagination(DEFAULT_VM_PAGE_SIZE);
 
-    const hasGenerateSBOMAbility = useHasGenerateSBOMAbility();
+    const hasGenerateSbomAbility = useHasGenerateSbomAbility();
     const isScannerV4Enabled = useIsScannerV4Enabled();
+    const [sbomTargetImage, setSbomTargetImage] = useState<string>();
```

Comment:
Any reason for `useState<string>()` versus `useState('')`

Reply:
In this case, I wanted the absence of string to represent the case where no image is selected for the modal. Using `''` would likely work as well, but I preferred a more strict separation between `string` and `undefined`.

Reply:
Sure, it surprised me that hook types did not require `<string | undefined>`

Reply:
Same at first - it looks like the typing is set up for convenience to not have to explicitly list `| undefined` each time:

```ts
function useState<S = undefined>(): [S | undefined, Dispatch<SetStateAction<S | undefined>>];
```